charmdir.go

```diff
@@ -534,6 +534,15 @@ func (dir *CharmDir) MaybeGenerateVersionString(logger Logger) (string, string,
 	vcsOrder := [...]string{"git", "hg", "bzr"}
 	cmdWaitTime := 2 * time.Second
 
+	absPath := dir.Path
+	if !filepath.IsAbs(absPath) {
+		var err error
+		absPath, err = filepath.Abs(dir.Path)
+		if err != nil {
+			logger.Warningf("failed resolving relative path %q", dir.Path)
```

Comment:
We don't want to fail here, as this would change the long-running behavior we had until now imo

Reply:
So, what happens if this fails and you fall through? Won't it cause issues with the code below? I think that absPath will be set to `""`

Reply:
Hmm you are right, i suppose fallback to path would be the way to have no regression?

Reply:
returning the error now.